grid_search.py

```diff
@@ -65,15 +71,24 @@ def grid_search_next_run(
 
     # this is O(N) due to the O(1) complexity of individual hash lookups; previous implementation was O(N^2)
     remaining_params = list(all_param_values - param_values_seen)
+    n_remaining = len(remaining_params)
 
     if randomize_order:
         random.shuffle(remaining_params)
 
-    # we have searched over the entire parameter space
-    if len(remaining_params) == 0:
-        return None
-
-    for param, value in zip(discrete_params, remaining_params[0]):
-        param.value = value
-
-    return SweepRun(config=discrete_params.to_config())
+    retval: List[Optional[SweepRun]] = []
+    for i in range(n):
+        if i < n_remaining:
```

Comment:
This is a hard to understand loop. took me a couple minutes to convince myself it handled all the cases.

I think this is cleaner:
```
for i in range(min(n, n_remaining)):
  ...
if n > n_remaining:
  retval.append(None)
```

Reply:
done